### PR TITLE
Add generic instances for two stage conv bwd wei

### PR DESCRIPTION
**Pull request by bartekxk**

### CODE REVIEW

File: library/include/ck/library/tensor_operation_instance/gpu/grouped_conv_bwd_weight/device_grouped_conv_bwd_weight_two_stage_xdl_instance.hpp [hunk 1]
```diff
@@ -32,6 +32,23 @@ static constexpr auto ConvBwdWeightDefault =
...
     ck::tensor_operation::device::ConvolutionBackwardWeightSpecialization::Filter1x1Stride1Pad0;
 
+template <ck::index_t NDimSpatial,
+          typename ALayout,
+          typename BLayout,
+          typename ELayout,
+          ConvolutionBackwardWeightSpecialization ConvSpec,
+          BlockGemmPipelineScheduler Scheduler,
+          BlockGemmPipelineVersion PipelineVersion>
+using device_grouped_conv_bwd_weight_two_stage_xdl_c_shuffle_f16_generic_instances = std::tuple<
```
Comment (aosewski) [inline library/include/ck/library/tensor_operation_instance/gpu/grouped_conv_bwd_weight/device_grouped_conv_bwd_weight_two_stage_xdl_instance.hpp]: Minor suggestion:
```suggestion
using device_grouped_conv_bwd_weight_two_stage_xdl_cshuffle_f16_generic_instances = std::tuple<
```

Comment (aosewski) [reply library/include/ck/library/tensor_operation_instance/gpu/grouped_conv_bwd_weight/device_grouped_conv_bwd_weight_two_stage_xdl_instance.hpp]: Please add the layout to the name.

File: library/include/ck/library/tensor_operation_instance/gpu/grouped_conv_bwd_weight/device_grouped_conv_bwd_weight_two_stage_xdl_instance.hpp [hunk 2]
```diff
@@ -57,6 +74,23 @@ using device_grouped_conv_bwd_weight_two_stage_xdl_c_shuffle_f16_instances = std
...
     >;
 
+template <ck::index_t NDimSpatial,
+          typename ALayout,
+          typename BLayout,
+          typename ELayout,
+          ConvolutionBackwardWeightSpecialization ConvSpec,
+          BlockGemmPipelineScheduler Scheduler,
+          BlockGemmPipelineVersion PipelineVersion>
+using device_grouped_conv_bwd_weight_two_stage_xdl_c_shuffle_bf16_generic_instances = std::tuple<
```
Comment (aosewski) [inline library/include/ck/library/tensor_operation_instance/gpu/grouped_conv_bwd_weight/device_grouped_conv_bwd_weight_two_stage_xdl_instance.hpp]: This instance parameters related to memory transfers seem to me identical as the one in: device_grouped_conv_bwd_weight_two_stage_ngchw_xdl_c_shuffle_bf16_generic_instances.  Does it mean it's the same layout or what's the difference ?

**Review by aosewski — COMMENTED** [review]
I have some confusion related to instances layout.

File: library/include/ck/library/tensor_operation_instance/gpu/grouped_conv_bwd_weight/device_grouped_conv_bwd_weight_two_stage_xdl_instance.hpp (hunk 1 above)
Comment (bartekxk) [reply library/include/ck/library/tensor_operation_instance/gpu/grouped_conv_bwd_weight/device_grouped_conv_bwd_weight_two_stage_xdl_instance.hpp]: Added.

File: library/include/ck/library/tensor_operation_instance/gpu/grouped_conv_bwd_weight/device_grouped_conv_bwd_weight_two_stage_xdl_instance.hpp (hunk 2 above)
Comment (bartekxk) [reply library/include/ck/library/tensor_operation_instance/gpu/grouped_conv_bwd_weight/device_grouped_conv_bwd_weight_two_stage_xdl_instance.hpp]: Can we investigate this in seperate PR?

**Review by aosewski — APPROVED** [review]

**Pull request merged**